apps/server/src/modules/tool/context-external-tool/service/context-external-tool.service.ts

```diff
@@ -36,15 +46,33 @@ export class ContextExternalToolService {
 		await this.contextExternalToolRepo.delete(contextExternalTools);
 	}
 
-	async deleteContextExternalTool(contextExternalTool: ContextExternalTool): Promise<void> {
+	public async deleteContextExternalTool(contextExternalTool: ContextExternalTool): Promise<void> {
 		await this.contextExternalToolRepo.delete(contextExternalTool);
 	}
 
-	async findAllByContext(contextRef: ContextRef): Promise<ContextExternalTool[]> {
+	public async findAllByContext(contextRef: ContextRef): Promise<ContextExternalTool[]> {
 		const contextExternalTools: ContextExternalTool[] = await this.contextExternalToolRepo.find({
 			context: contextRef,
 		});
 
 		return contextExternalTools;
 	}
+
+	public async checkContextRestrictions(
+		contextExternalTool: ContextExternalTool,
+		userId: EntityId,
+		context: AuthorizationContext
+	): Promise<void> {
+		const schoolExternalTool: SchoolExternalTool = await this.schoolExternalToolService.findById(
+			contextExternalTool.schoolToolRef.schoolToolId
+		);
+
+		const externalTool: ExternalTool = await this.externalToolService.findById(schoolExternalTool.toolId);
+
+		if (externalTool.restrictToContexts && externalTool.restrictToContexts[0]) {
+			if (!externalTool.restrictToContexts.includes(contextExternalTool.contextRef.type)) {
+				throw new ForbiddenLoggableException(userId, 'ContextExternalTool', context);
```

Comment:
I would suggest bad request instead, because the user input is wrong

Reply:
Bad request is used for malformed request syntax (e.g. if you write something like the context wrong, "CouRsE" instead of "course") but the syntax is correct in this case. UnporcessableEntity means that the server understands the content type and the syntax is correct, but it could not fully execute because of semantic issues.